scheds/rust/scx_layered/src/main.rs

```diff
@@ -1966,54 +1970,78 @@ impl<'a> Scheduler<'a> {
                 updated = true;
             }
         }
+        Ok(updated)
+    }
 
-        // Give the rest to the open layers.
-        if updated {
-            for (idx, layer) in self.layers.iter_mut().enumerate() {
-                if !layer_is_open(layer) {
-                    continue;
-                }
-
-                let bpf_layer = &mut self.skel.maps.bss_data.layers[idx];
-                let available_cpus = self.cpu_pool.available_cpus_in_mask(&layer.allowed_cpus);
-                let nr_available_cpus = available_cpus.count_ones();
+    /// Refreshes cpumasks on the bpf side.
+    fn refresh_cpumasks_bpf(&mut self) {
+        let input = ProgramInput {
+            ..Default::default()
+        };
+        let prog = &mut self.skel.progs.refresh_layer_cpumasks;
+        let _ = prog.test_run(input);
+    }
 
-                // Open layers need the intersection of allowed cpus and
-                // available cpus.
-                layer.cpus.copy_from_bitslice(&available_cpus);
-                layer.nr_cpus = nr_available_cpus;
-                Self::update_bpf_layer_cpumask(layer, bpf_layer);
+    /// Distributes remaining CPUs across open layers.
+    fn grow_open_layers(&mut self) -> Result<()> {
```

Comment:
This is a misnomer. The function doesn't just grow open layers. It also performs all the post-update operations including triggering BPF side updates and recording layer cpus_range and so on. It also can shrink open layers -ie. the right verb would be something like `fit`. It probably would make sense to split this into two separate functions.